File: algorithms/1684.count-the-number-of-consistent-strings/solution.go
func countConsistentStrings(allowed string, words []string) int {
    count := 0

    for _, word := range words {
        onlyAllowedChars := true

        for _, c := range word {
            if !strings.ContainsRune(allowed, c) {
                onlyAllowedChars = false
                break
            }
        }

        if onlyAllowedChars {
            count += 1
        }
    }

    return count
}

File: algorithms/1684.count-the-number-of-consistent-strings/solution.rb
# @param {String} allowed
# @param {String[]} words
# @return {Integer}
def count_consistent_strings(allowed, words)
  count = 0
  allowed_chars = allowed.chars

  words.each do |word|
    only_allowed_chars = true
    word_chars = word.each_char.sort.uniq

    word_chars.each do |c|
      unless allowed_chars.include?(c)
        only_allowed_chars = false
        break
      end
    end
    count += 1 if only_allowed_chars
  end

  count
end
